syntax = "proto3";

package fbp;

service CRMasterServer { 
    rpc Connect (Message) returns (Reply) {}
    rpc Login (Message) returns (Reply) {} 
    rpc List (Message) returns (ListReply) {} 
    rpc Join (Message) returns (Reply) {}
    rpc Leave (Message) returns (Reply) {}
    rpc Chat (stream Message) returns (stream Message) {} 
	rpc Cast (Message) returns () {}
} 

message ListReply { 
    repeated string all_roomes = 1; 
    repeated string joined_roomes = 2; 
} 

message Message { 
    string username = 1; 
    string msg = 2; 
}

message Reply { 
    string msg = 1; 
}
